python/ray/tests/test_task_metrics.py

```diff
@@ -456,19 +456,22 @@ def f(x):
 ray.get([f.remote(x) for x in buf])"""
 
     proc = run_string_as_driver_nonblocking(driver)
-    expected = {
-        "RUNNING": 2.0,
-        "PENDING_ARGS_FETCH": 7.0,
-        "PENDING_OBJ_STORE_MEM_AVAIL": 91.0,
-    }
+
+    # TODO(ekl) why is LocalTaskManager nondeterministic in how many tasks end up in
+    # the dispatch queue? Otherwise, we should expect 7 pending node assignment.
```

Comment:
This could be a bug in LocalTaskManager--- sometimes, there are much more tasks in the dispatch state than you would expect could be possible given the object store memory size.